### DISCUSSION TITLE
2023/04/02/contract-first-openapi-development-but-still-use-swagger-ui-with-asp.net-core/

### DISCUSSION THREAD

**Discussion by giscus** [CodingMilitia/Blog#91] [Blog comments]
# 2023/04/02/contract-first-openapi-development-but-still-use-swagger-ui-with-asp.net-core/

In C# and .NET land, we’re pretty heavy on the code first approaches, with the odd exception. Let’s take a look at a possible contract first approach to API development, with OpenAPI, but still taking advantage of existing tooling that we’ve come to rely on, like Swagger UI.

https://blog.codingmilitia.com/2023/04/02/contract-first-openapi-development-but-still-use-swagger-ui-with-asp.net-core/

**Comment by srpeirce** [comment]
Great article, thanks for sharing. 

As you mentioned in the article, I've seen contracts break but the tests still pass as they are using the same model (e.g. Somebody renames a typo in a property) - this seems a good way of avoiding that.

How is maintainability over time?

**Reply by joaofbantunes** [reply]
Hey, thanks!

I can't speak from experience about maintainability, as I haven't been using this approach (yet). The post was triggered by me noticing those issues, and trying to find something better.

In any case, I wouldn't expect the long term maintainability to be harder than the typical code-first approach. It might be a bit more tedious, as we need to adjust the contract and the code separately. Despite more tedious, I would actually expect it to be simpler, in particular for more complex APIs, as we avoid trying to make magic happen with the contract generator 🙂.
